tests/RequestTest.php

```diff
@@ -97,6 +97,29 @@ public function testSameInstanceWhenSameUri()
         $this->assertSame($r1, $r2);
     }
 
+    public function testWithUriParameterPreserveHost()
```

Comment:
please make each case a different test case with concrete function names that it tests